src/lib/y2network/interface.rb

```diff
@@ -49,6 +52,19 @@ class Interface
       end
     end
 
+    # Creates an instance of interface class based on the given interface type
+    #
+    # @param name [String] interface name (eth0, ...)
+    # @param type [Y2Network::InterfaceType] interface type (InterfaceType::ETHERNET, ...)
+    def self.for(name, type)
+      type_name = type.short_name
+      require "y2network/interfaces/#{type_name}_interface"
+      Interfaces.const_get(type_name.to_s.capitalize + "Interface").new(name, type: type)
```

Comment:
I suggest to wait a bit or base it on my interface_type pr, which adds methods for file_name and class_name. Also it is inconsistent to others which does not have suffix as namespace is enough. So ideally it should be Ethernet class and not EthernetInterface class.

Reply:
also I think it would be nice to support also type as short name string.